MapboxNavigation/InstructionsBannerViewLayout.swift

```diff
@@ -119,15 +119,11 @@ extension BaseInstructionsBannerView {
     func setupAvailableBounds() {
         // Abbreviate if the instructions do not fit on one line
         primaryLabel.availableBounds = {
-            let height = ("|" as NSString).size(withAttributes: [.font: self.primaryLabel.font]).height
-            let availableWidth = self.bounds.width-self.maneuverView.frame.maxX-(8*2)
-            return CGRect(x: 0, y: 0, width: availableWidth, height: height)
+            return CGRect(x: 0, y: 0, width: self.bounds.width, height: self.primaryLabel.font.lineHeight)
```

Comment:
This is the reason for all those test fixture changes down below. In effect, the label now extends to the maximum trailing margin instead of stopping at the natural text boundary. That’s typically the best practice for accommodating right-to-left text. Otherwise, the label ended up being abbreviated far too aggressively.

Reply:
I don't think abbreviations will work correctly when using the full width here.
The `availableWidth` should be equal to `bounds.width-maneuverView.bounds.width-BaseInstructionsBannerView.padding*2`.

Reply:
In left-to-right layout, at least, it seems to have the same behavior as on master:

Before | After
----|----
<img src="https://user-images.githubusercontent.com/1231218/36910521-dab2b83a-1df5-11e8-8c1b-207fb9ff868f.png" width="200" alt="foothill"> | <img src="https://user-images.githubusercontent.com/1231218/36910666-39513920-1df6-11e8-8d99-4a44242508b7.png" width="200" alt="foothill">
<img src="https://user-images.githubusercontent.com/1231218/36910444-a2726cae-1df5-11e8-915a-5e0e8ebe7f7a.png" width="200" alt="glen este"> | <img src="https://user-images.githubusercontent.com/1231218/36909734-bfac406c-1df3-11e8-9c65-246271a3ca53.png" width="200" alt="glen este-withamsville">
<img src="https://user-images.githubusercontent.com/1231218/36910342-4788f2e0-1df5-11e8-947f-5926d14811ed.png" width="200" alt="mlk"> | <img src="https://user-images.githubusercontent.com/1231218/36910186-dc2d2e76-1df4-11e8-9f17-c9105e96dc71.png" width="200" alt="mlk">

Reply:
> I used `maxX` to simplify the following availableWidth format: 
`H:|-18-maneuverView-18-availableWidth-18-|`
so that the left side padding was accounted for by the maneuverView’s`maxX` value.

Doesn’t the maneuver view already get excluded from the label’s bounds?

Reply:
Hmm, `self.bounds` here refers to `InstructionsBannerView.bounds` no? 🤔 
Using `primaryLabel.bounds.width` might work now when we changed from a `lessThanOrEqualTo` constraint to an `equalTo` constraint.

Reply:
Using `primaryLabel` et al. seems to abbreviate too aggressively when the component gets the whole label width:

Expected | Actual
----|----
<img width="414" alt="reference_testprimaryshieldandsecondary 2x" src="https://user-images.githubusercontent.com/1231218/36911991-2d98cd10-1dfa-11e8-8d8b-d8302684e528.png"> | <img width="414" alt="failed_testprimaryshieldandsecondary 2x" src="https://user-images.githubusercontent.com/1231218/36912000-34d50bde-1dfa-11e8-90f6-629232450591.png">
<img width="414" alt="reference_testsinglelineprimaryandsecondary 2x" src="https://user-images.githubusercontent.com/1231218/36912012-3bc34082-1dfa-11e8-8fbd-06f829c008a1.png"> | <img width="414" alt="failed_testsinglelineprimaryandsecondary 2x" src="https://user-images.githubusercontent.com/1231218/36912017-3ef77796-1dfa-11e8-86e3-88167bc05f65.png">

Reply:
8465018814ba78e72ba04158ec430c725aeea1dc pares back this change. I didn’t realize originally that `self` was InstructionBannerView rather than the primary or secondary label, but of course that’s the case. The new code switches on the interface direction to avoid overabbreviating in right-to-left layout.